index.js

```diff
@@ -4,18 +4,41 @@ const { getApolloServer } = require("./apollo");
 
 (async () => {
   const app = express();
-  app.use(express.json({ limit: "2mb" }));
+  const jsonBodyLimit = getENV("GRAPHQL_JSON_BODY_LIMIT", "2mb");
+  app.use(express.json({ limit: jsonBodyLimit }));
 
   const server = await getApolloServer();
-  server.applyMiddleware({ app });
+  let middleware = server.getMiddleware({});
+  app.use((req, res, next) => {
+    middleware(req, res, next);
+  });
+
+  const activateUpdateGatewayInterval = getENV("GRAPHQL_UPDATE_GATEWAY", "false") === "true";
+  const updateGatewayInterval = getENV("GRAPHQL_UPDATE_GATEWAY_INTERVAL_MS", "60000");
+  if (activateUpdateGatewayInterval === true)
+  {
+    setInterval(async () => {
+      // console.log("Updating graphql gateway..");
+      try {
+        const server = await getApolloServer();
+        let middleware = server.getMiddleware({});
+        app.use((req, res, next) => {
+          middleware(req, res, next);
+        });
```

Comment:
```suggestion
        middleware = server.getMiddleware({});
```

Reply:
Is there a need to use `app.use`? Shouldn't be updating the middleware instance be enough?

Reply:
No, that's correct. I'll change it